File: Problem/BOJ_10814_나이순 정렬/main.py
from sys import stdin


def main():
    stdin = open('./input.txt', 'r')
    num_of_members = int(stdin.readline())

    members = []
    for idx in range(num_of_members):
        member_info = stdin.readline().split()
        members.append([idx] + [int(member_info[0])] + [member_info[1]])

    members = sorted(members, key=lambda x: (x[1], x[0]))

    for _, age, name in members:
        print(str(age) + " " + name)


if __name__ == '__main__':
    main()

File: Problem/BOJ_10814_나이순 정렬/main.js
const fs = require('fs')
const input = fs.readFileSync('./input.txt').toString().split("\n")


const numOfMembers = parseInt(input[0])
const members = input.slice(1).map((member, idx) => {
    const [age, name] = member.split(" ")
    return [idx, parseInt(age), name]
})

members.sort((a, b) => {
    if (a[1] < b[1]) {
        return -1
    } else if (a[1] > b[1]) {
        return 1
    } else if (a[1] === b[1]) {
        return a[0] - b[0]
    }
})

for (let i = 0; i < numOfMembers; i++) {
    console.log(`${members[i][1]} ${members[i][2]}`)
}